### PR TITLE
Fix SslStream drop bug when create by from_raw_parts function

### PR DESCRIPTION
**Pull request by vkill**
Hi there

Please review it, thank you.

### CODE REVIEW

File: openssl/src/ssl/mod.rs [hunk 1]
```diff
@@ -3483,7 +3487,9 @@ impl<S: Read + Write> SslStream<S> {
     /// The caller must ensure the pointer is valid.
     pub unsafe fn from_raw_parts(ssl: *mut ffi::SSL, stream: S) -> Self {
```
Comment (sfackler) [inline openssl/src/ssl/mod.rs:3488]: This method needs to take ownership of the SSL.

Comment (vkill) [reply openssl/src/ssl/mod.rs:3488]: The following is an example, maybe the scene is wrong, the ssl ptr should from ffi. is that what you mean?

```rust
use std::io;
use std::net::TcpStream;

use foreign_types::ForeignTypeRef;
use openssl::ssl::{SslConnector, SslMethod, SslStream, SslVerifyMode};

fn main() -> io::Result<()> {
    let tcp_stream = TcpStream::connect("echo.websocket.org:443")?;

    let mut ssl_connector_build = SslConnector::builder(SslMethod::tls()).unwrap();
    ssl_connector_build.set_verify(SslVerifyMode::NONE);
    let ssl_connector = ssl_connector_build.build();
    let ssl_stream = ssl_connector
        .connect("echo.websocket.org", tcp_stream)
        .unwrap();

    let ssl_ptr = ssl_stream.ssl().as_ptr();
    let tcp_stream = ssl_stream.get_ref();

    let _ = unsafe { SslStream::from_raw_parts(ssl_ptr, tcp_stream) };

    Ok(())
}

```

Comment (sfackler) [reply openssl/src/ssl/mod.rs:3488]: You cannot share an SSL between two streams.

Comment (vkill) [reply openssl/src/ssl/mod.rs:3488]: Oh, I see, thank you.

**Pull request closed without merging**